todo_rxbloc_app/lib/feature_todos_list/di/todos_list_dependecies.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2022, Prime Holding JSC
+// https://www.primeholding.com
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter_rx_bloc/flutter_rx_bloc.dart';
+import 'package:provider/provider.dart';
+import 'package:provider/single_child_widget.dart';
+
+import '../../base/routers/router.dart';
+import '../bloc/todos_list_bloc.dart';
+
+class TodosListDependecies {
+  TodosListDependecies._(this.context);
+
+  factory TodosListDependecies.of(BuildContext context) => _instance != null
+      ? _instance!
```

Comment:
As we discussed you need to use `.from` instead of `.of` also the singleton pattern is not needed here, since it causes memory leaks